src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -79,868 +78,290 @@
-    ;; would be to split this out into two tables.
-    ["CONSTRAINT constraint_resource_events_unique
-       UNIQUE (report, resource_type, resource_title, property)"])
-
-   "CREATE INDEX idx_reports_certname ON reports(certname)"
-
-   ;; I presume we'll be doing a decent number of queries sorted by a timestamp,
-   ;; and this seems like the most likely candidate out of the timestamp fields
-   "CREATE INDEX idx_reports_end_time ON reports(end_time)"
-   "CREATE INDEX idx_resource_events_report ON resource_events(report)"
-   "CREATE INDEX idx_resource_events_resource_type ON resource_events(resource_type)"
-   "CREATE INDEX idx_resource_events_resource_type_title ON resource_events(resource_type, resource_title)"
-   "CREATE INDEX idx_resource_events_timestamp ON resource_events(timestamp)"))
-
-(defn add-parameter-cache
-  "Creates the new resource_params_cache table, and populates it using
-  the existing parameters in the database."
-  []
-  ;; Create cache table
-  (jdbc/do-commands
-   (sql/create-table-ddl :resource_params_cache
-                         ["resource" "VARCHAR(40)"]
-                         ["parameters" "TEXT"]
-                         ["PRIMARY KEY (resource)"]))
-
-  (log/warn "Building resource parameters cache. This make take a few minutes, but faster resource queries are worth it.")
-
-  ;; Loop over all parameters, and insert a cache entry for each resource
-  (let [query    ["SELECT resource, name, value from resource_params ORDER BY resource"]
-        collapse (fn [rows]
-                   (let [resource (:resource (first rows))
-                         params   (into {} (map #(vector (:name %) (json/parse-string (:value %))) rows))]
-                     [resource params]))]
-
-    (jdbc/with-query-results-cursor query
-      (fn [rs]
-        (let [param-sets (->> rs
-                              (partition-by :resource)
-                              (map collapse))]
-          (doseq [[resource params] param-sets]
-            (jdbc/insert! :resource_params_cache
-                          {:resource   resource
-                           :parameters (json/generate-string params)}))))))
-
-  ;; Create NULL entries for resources that have no parameters
-  (jdbc/do-commands
-   "INSERT INTO resource_params_cache
-    SELECT DISTINCT resource, NULL FROM catalog_resources WHERE NOT EXISTS
-    (SELECT 1 FROM resource_params WHERE resource=catalog_resources.resource)"
-
-   "ALTER TABLE catalog_resources ADD FOREIGN KEY (resource)
-      REFERENCES resource_params_cache(resource) ON DELETE CASCADE"
-   "ALTER TABLE resource_params ADD FOREIGN KEY (resource)
-      REFERENCES resource_params_cache(resource) ON DELETE CASCADE"))
-
-(defn add-event-status-index
-  "Add an index to the `status` column of the event table."
-  []
-  (jdbc/do-commands
-   "CREATE INDEX idx_resource_events_status ON resource_events(status)"))
+(defn init-through-2-y-z []
```

Comment:
Why not 3.0.0? We only support migrating from one version back, and master is going to be 4.x, right?